cmd/internal/printconfig/command_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package printconfig
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+)
+
+func TestCommand(t *testing.T) {
+	testPairs := []struct {
+		Key   string
+		Value string
+	}{
+		{"TEST_METRICS", "1"},
+		{"TEST_REPORTER_LOG_SPANS", "true"},
+		{"TEST_NUM_WORKS", "0"},
+	}
+	source := "user-assigned"
+
+	v := viper.New()
+
+	v.SetDefault(testPairs[0].Key, testPairs[0].Value)
+	v.SetDefault(testPairs[1].Key, testPairs[1].Value)
+	v.SetDefault(testPairs[2].Key, testPairs[2].Value)
+
+	maxSourceLength := len(source)
+	maxKeyLength, maxValueLength := 0, 0
+	for _, t := range testPairs {
+		if len(t.Key) > maxKeyLength {
+			maxKeyLength = len(t.Key)
+		}
+		if len(t.Value) > maxValueLength {
+			maxValueLength = len(t.Value)
+		}
+	}
```

Comment:
This is a bad practice for tests. What you're effectively doing is saying: I have production code that computes stuff. I am going to write nearly identical code in the test to verify that that other code is "correct".

A much better approach is to simply encode the final expected output (also known as snapshot testing). It makes the test very readable, as you can see exactly what is expected to be produced, and it doesn't have that circular fallacy of testing one code with similarly looking code. And then just do `assert.Equals(t, expected, actual)`